### DISCUSSION TITLE
Compare the flux generated by Human gem model

### DISCUSSION THREAD

**Discussion by studymeow** [SysBioChalmers/Human-GEM#243] [Q&A] [answered]
Thank you very much for developing such a great model and it's really helpful. I am new to FBA and I recently used this model to generate some results of flux balance analysis. 

I hope I could get some suggestion on the model results process. To be more specific, I want to compare the fluxes with experimental fluxes and I have read the human gem paper "atlas of human metabolism".  I encountered the following figure. 

![07678CA7-C1B8-4114-96FB-57A8948E1792](https://user-images.githubusercontent.com/58016194/112114487-1fc7d300-8b86-11eb-9855-3d9efd24593f.jpeg)

I was just wondering why would we want to take the absolute value of fluxes to compare given the directionality of fluxes does matter. Do we have to it to compare the two fluxes? Can we just look at the original fluxes to calculate the correlation measure? 

Thank you so much!

**Comment by haowang-bioinfo** [comment]
@studymeow thanks for your interests to our work.

A brief answer to your question: this comparison demonstrates the capability of `ecHuman` models in predicting nutrient uptake rates by evaluating with some experimental measurements.

**Comment by JonathanRob** [comment]
Good question, @studymeow. Since I think it's more fitting as a Discussion rather than an Issue, I will convert this Issue to a Discussion.

**Comment by JonathanRob** [comment, marked as answer]
You raise some excellent points, @studymeow. There is not always an easy way to compare fluxes in the most informative manner. For example, we considered plotting the actual values directly and calculating a correlation as you suggest; however, since the fluxes vary by quite large amounts (orders of magnitude in some cases), the large fluxes dominated and resulted in an inflated correlation coefficient that did not faithfully represent the data.

We therefore decided to log-transform the fluxes to place more weight on the lower magnitude fluxes. Unfortunately, this of course means we cannot use the values directly since many values are negative. We therefore compared the log-transformed absolute flux values, noting that the sign of the fluxes were in agreement for all cases except a very small fraction which anyway had very low magnitudes.

Another option could be to simply calculate the percent or relative error for each of the fluxes. However, we again run into problems here when the magnitude of the fluxes are very low, as even tiny absolute differences will result in very large relative differences.

I'm sure there must be some better ways to compare fluxes that others have developed, so please let us know if you find a good method that can address these challenges! I'm sure other users (including us) would find it quite helpful.

**Reply by studymeow** [reply]
Thank you very much for such a great and detailed answer. I indeed found the correlation coefficient was inflated when comparing the original fluxes value. I will explore more on this issue. Thank you!
